python/plugins/processing/algs/qgis/ZonalStatisticsQgis.py

```diff
@@ -75,24 +75,23 @@ def __init__(self):
                       self.tr('All'): QgsZonalStatistics.All
                       }
 
-        self.addParameter(ParameterRaster(self.INPUT_RASTER,
-                                          self.tr('Raster layer')))
-        self.addParameter(ParameterNumber(self.RASTER_BAND,
-                                          self.tr('Raster band'),
-                                          1, 999, 1))
-        self.addParameter(ParameterVector(self.INPUT_VECTOR,
-                                          self.tr('Vector layer containing zones'),
-                                          [dataobjects.TYPE_VECTOR_POLYGON]))
-        self.addParameter(ParameterString(self.COLUMN_PREFIX,
-                                          self.tr('Output column prefix'), '_'))
-        self.addParameter(ParameterSelection(self.STATISTICS,
-                                             self.tr('Statistics to calculate'),
-                                             list(self.STATS.keys()),
-                                             multiple=True))
-        self.addOutput(OutputVector(self.OUTPUT_LAYER,
-                                    self.tr('Zonal statistics'),
-                                    True,
-                                    datatype=[dataobjects.TYPE_VECTOR_POLYGON]))
+        self.addParameter(QgsProcessingParameterRasterLayer(self.INPUT_RASTER,
+                                                            self.tr('Raster layer')))
+        self.addParameter(QgsProcessingParameterNumber(self.RASTER_BAND,
+                                                       self.tr('Raster band'),
+                                                       minValue=1, maxValue=999, defaultValue=1))
+        self.addParameter(QgsProcessingParameterVectorLayer(self.INPUT_VECTOR,
+                                                            self.tr('Vector layer containing zones'),
+                                                            [dataobjects.TYPE_VECTOR_POLYGON]))
```

Comment:
The dataobjects.* enum values here need flipping to their equivalents from QgsProcessingParameterDefinitions